src/network/networkd-link.c

```diff
@@ -40,6 +40,20 @@ DUID* link_get_duid(Link *link) {
                 return &link->manager->duid;
 }
 
+static bool link_sysctl_ipv6_enabled(Link *link) {
+        _cleanup_free_ char *value = NULL;
```

Comment:
```link_enable_ipv6``` we use to enable or disable the sysctl . seems like this function is reductant .

Reply:
`link_enable_ipv6()` write the value, while added function reads the value...
It may be possible to optimize the flow, but I'd like to leave that in another PR.

Reply:
if  ```link_enable_ipv6``` will be called afterwards it will change the settings depending on the conf and it returns the value what is has written in the conf.

Reply:
Yeah, so `sysctl_ipv6_enabled` is set to `-1`. Is it not sufficient?

Reply:
now sure how that matters. Because the sequence of calling what really matters here.